homeassistant/components/tesla_fleet/oauth.py

```diff
@@ -0,0 +1,77 @@
+"""Application Credentials platform the Tesla Fleet integration."""
+
+import base64
+import hashlib
+import secrets
+from typing import Any
+
+from homeassistant.components.application_credentials import (
+    AuthImplementation,
+    AuthorizationServer,
+    ClientCredential,
+)
+from homeassistant.core import HomeAssistant
+
+from .const import AUTHORIZE_URL, CLIENT_ID, DOMAIN, SCOPES, TOKEN_URL
+
+AUTH_SERVER = AuthorizationServer(AUTHORIZE_URL, TOKEN_URL)
+
+
+class TeslaSystemImplementation(AuthImplementation):
```

Comment:
We should inherit the `LocalOAuth2Implementation` class here and not the application credentials class since we're not going to use this implementation with application credentials.